shakenfist/daemons/net.py

```diff
@@ -18,77 +18,90 @@ class monitor(object):
+
+        # Ensure we are on every network we have a host for
+        for network in host_networks:
+            with db.get_lock('sf/network/%s' % network['uuid'], ttl=120) as _:
                 n = net.from_db(network)
+                if not n:
+                    continue
+                if n['state'] != 'created':
+                    continue
+
                 n.create()
                 n.ensure_mesh()
                 seen_vxids.append(n.vxlan_id)
 
-            # Determine if there are any extra vxids
-            extra_vxids = list(vxid_to_mac.keys())
-            for seen in seen_vxids:
-                if seen in extra_vxids:
-                    extra_vxids.remove(seen)
-
-            # For now, just log extra vxids
-            if extra_vxids:
-                LOG.warn('Extra vxlans present! IDs are: %s'
-                         % extra_vxids)
-
-                # Determine the network uuids for those vxids
-                vxid_to_uuid = {}
-                for n in db.get_networks():
-                    vxid_to_uuid[n['vxid']] = n['uuid']
-
-                for extra in extra_vxids:
-                    if extra in vxid_to_uuid:
+        # Determine if there are any extra vxids
+        extra_vxids = list(vxid_to_mac.keys())
+        for seen in seen_vxids:
+            if seen in extra_vxids:
+                extra_vxids.remove(seen)
+
+        # For now, just log extra vxids
+        if extra_vxids:
+            LOG.warn('Extra vxlans present! IDs are: %s'
+                     % extra_vxids)
+
+            # Determine the network uuids for those vxids
+            vxid_to_uuid = {}
+            for n in db.get_networks():
+                vxid_to_uuid[n['vxid']] = n['uuid']
+
+            for extra in extra_vxids:
+                if extra in vxid_to_uuid:
+                    with db.get_lock('sf/network/%s' % vxid_to_uuid[extra], ttl=120) as _:
                         n = net.from_db(vxid_to_uuid[extra])
                         n.delete()
                         LOG.info('Extra vxlan %s (network %s) removed.'
                                  % (extra, vxid_to_uuid[extra]))
-                    else:
-                        LOG.error('Extra vxlan %s does not map to any network.'
-                                  % extra)
+                else:
+                    LOG.error('Extra vxlan %s does not map to any network.'
+                              % extra)
+
+        # And record vxids in the database
+        db.persist_node_vxid_mapping(
+            config.parsed.get('NODE_NAME'), vxid_to_mac)
+
+    def run(self):
+        while True:
+            time.sleep(30)
 
-            # And record vxids in the database
-            db.persist_node_vxid_mapping(
-                config.parsed.get('NODE_NAME'), vxid_to_mac)
+            try:
+                self._maintain_networks()
+            except Exception as e:
+                LOG.error('Ingored exception in network monitor: %s' % e)
```

Comment:
"ingored"

Should these be labelled so that Sentry (or whatever) can find them more easily?
* Prefixed with "sf-monitor" label or something
* Some sort of label like "Exception: "

Reply:
Done.